pre-commit

```diff
@@ -76,59 +76,68 @@ function checkyaml() {
 
 setup_paths
 
-echo "### Checking puppet syntax, for science! ###"
-# for file in $(git diff --name-only --cached | grep -E '\.(pp|erb)')
-for file in $(git diff --name-only --cached | grep -E '\.(pp)'); do
-  # Only check new/modified files that end in *.pp extension
-  if [[ -f $file && $file == *.pp ]]; then
-    $path_to_puppet_lint \
-      --no-80chars-check \
-      --no-autoloader_layout-check \
-      --no-nested_classes_or_defines-check \
-      --no-class_inherits_from_params_class-check \
-      --with-filename $file
-
-    # Set us up to bail if we receive any syntax errors
-    if [[ $? -ne 0 ]]; then
-      syntax_is_bad=1
-    else
-      echo "$file looks good"
+if [[ $(git diff --name-only --cached | grep -E '\.(pp)') ]]; then
+  echo "### Checking puppet syntax, for science! ###"
+  # for file in $(git diff --name-only --cached | grep -E '\.(pp|erb)')
+  for file in $(git diff --name-only --cached | grep -E '\.(pp)'); do
```

Comment:
I like this approach, gives a very clean result. Will merge it like this, thank you!